src/App.js

```diff
@@ -37,14 +41,54 @@ class App extends Component {
       })
   }
 
+  selectGem = (id) => {
+   let selectedGem = this.state.gems.find((gem) => {
+      return gem.id === id
+    })
+    this.setState({
+      selectedGem
+    })
+    this.changeDisplayPage('metals')
+  }
+
+  selectMetal = (id) => {
+    let selectedMetal = this.state.metals.find((metal) => {
+      return metal.id === id
+    })
+    this.setState({
+      selectedMetal
+    })
+    this.changeDisplayPage('results')
+  }
```

Comment:
Hmm, these two methods `selectGem` and `selectMethod` look pretttttyyyyy similar to me. Any way we could pass in an extra argument to combine them into one, more dynamic method??